doc/whats-new.rst

```diff
@@ -33,11 +33,13 @@ New Features
 
 Breaking changes
 ~~~~~~~~~~~~~~~~
-
+- Warn instead of raise when phony_dims are detected when using :py:func:`open_datatree` for h5netcdf-backend(:issue:`10049`, :pull:`10058`)
```

Comment:
```suggestion
- Warn instead of raise if phony_dims are detected when using :py:func:`open_datatree` for h5netcdf-backend and ``phony_dims=None`` (:issue:`10049`, :pull:`10058`)
```